Libplanet/Net/Swarm.cs

```diff
@@ -436,6 +432,77 @@ IEnumerator IEnumerable.GetEnumerator()
             }
         }
 
+        private async Task ProcessMessageAsync<T>(
+            Blockchain<T> blockchain,
+            Message message,
+            CancellationToken cancellationToken)
+            where T : IAction
+        {
+            switch (message)
```

Comment:
Couldn't this switch be restructured so that the `Message` subtypes has a polymorphic (virtual) method to process the equivalent thing?

Reply:
I've tried replace with a sort of `Prcoessor` or `Message.Process()`, but there are also tricky points.

- Some message (e.g. `PeerSetDelta`) requires `Swarm`'s internal properties. (`_deltas`, `_peers`)
- (`Message.Process()`) I'm not sure that it is correct to deal with the logic of how a message is processed, rather than simply representing the data.

Reply:
My concern is that such function, which consists of a `switch` branch, is prone to miss cases when a new case (e.g., a new `Message` subtype) is added, which leads it to be [partial][1] not to try.  Unfortunately C# does not warn even if a `switch` is not total.  On the other hand, making a method `virtual` and leaving a subtype to not implement it, of course, is disallowed by C#.  We can be safe from a new case without its case handling.  For that reason, I believe it's better to be polymorphic and should be eventually, even though we could leave it as is in this PR.

[1]: https://wiki.haskell.org/Partial_functions